src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -705,22 +710,23 @@ private static ConcurrentQueue<BlockCommitSet> CreateQueueAtomic(ref ConcurrentQ
             return prior ?? instance;
         }
 
-        private void CreateCommitSet(long blockNumber)
+        private BlockCommitSet CreateCommitSet(long blockNumber)
         {
             if (_logger.IsDebug) _logger.Debug($"Beginning new {nameof(BlockCommitSet)} - {blockNumber}");
 
             // TODO: this throws on reorgs, does it not? let us recreate it in test
-            Debug.Assert(CurrentPackage is null || blockNumber == CurrentPackage.BlockNumber + 1, "Newly begun block is not a successor of the last one");
-            Debug.Assert(IsCurrentListSealed, "Not sealed when beginning new block");
+            Debug.Assert(!(_commitSetQueue ?? CreateQueueAtomic(ref _commitSetQueue)).TryDequeue(out BlockCommitSet lastSet) || blockNumber == lastSet.BlockNumber + 1, $"Newly begun block is not a successor of the last one.");
+            Debug.Assert(!(_commitSetQueue ?? CreateQueueAtomic(ref _commitSetQueue)).TryDequeue(out lastSet) || lastSet.IsSealed, "Not sealed when beginning new block");
 
             BlockCommitSet commitSet = new(blockNumber);
             (_commitSetQueue ?? CreateQueueAtomic(ref _commitSetQueue)).Enqueue(commitSet);
             LatestCommittedBlockNumber = Math.Max(blockNumber, LatestCommittedBlockNumber);
+            // Why are we announcing **before** committing next block??
+            // Should it be after commit?
```

Comment:
Is this event now used outside of tests? Maybe it can be deleted?

Reply:
Used by TrieStoreBoundaryWatcher to update BestPersistedState.

Reply:
Ok then I think it can be done after commit